### PR TITLE
Scan/MassList sorting, filtering, similarity, jdoc

### PR DESCRIPTION
**Pull request by robinschmid**
Hey Tomas, 
Hey Kai,

this PR is just to highlight some classes/methods that might be helpful to the MS/MS merging or redundant after PR #568 .

Cheers
Robin

### CODE REVIEW

File: src/main/java/net/sf/mzmine/util/maths/similarity/Similarity.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import net.sf.mzmine.util.maths.Transform;
 
 /**
- * Different similarity measures
+ * Different similarity measures such as COSINE SIMILARITY, PEARSON, SPEARMAN,...
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/util/maths/similarity/Similarity.java]: Similarity class to collect many different similarity measures and some util methods

File: src/main/java/net/sf/mzmine/util/maths/similarity/SpectraSimilarity.java [hunk 2]
```diff
@@ -55,6 +54,7 @@ public double getCosine() {
 
 
   /**
+   * Cosine similarity
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/util/maths/similarity/SpectraSimilarity.java]: SpectraSimilarity

Create similarity object for two scans (masslist -> DataPoint[])

File: src/main/java/net/sf/mzmine/util/scans/ScanUtils.java [hunk 3]
```diff
@@ -621,7 +621,7 @@ public static int findBestFragmentScan(RawDataFile dataFile, Range<Double> rtRan
    * Sorted list (best first) of all MS2 fragmentation scans with n signals >= noiseLevel in the
    * specified or first massList, if none was specified
    * 
-   * @param row
+   * @param row all MS2 scans of all features in this row
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/util/scans/ScanUtils.java]: ScanUtils

getMassListOrFirst to always work on masslists (the user can supply a masslist name or it uses the first masslist)

listAllFragmentScans(row, sortingMode)

File: src/main/java/net/sf/mzmine/util/scans/sorting/ScanSortMode.java [hunk 4]
```diff
@@ -18,6 +18,12 @@
 
 package net.sf.mzmine.util.scans.sorting;
 
+/**
+ * Sort all scans (-> masslists -> DataPoint[]) by MAX_TIC or NUMBER_OF_SIGNALS
```
Comment (robinschmid) [inline src/main/java/net/sf/mzmine/util/scans/sorting/ScanSortMode.java]: ScanSortMode and ScanSorter and MasslistSorter

Collection of sorting modes. 
The ScanSorter always works on MassLists

**Comment by tomas-pluskal** [comment]
Thanks, Robin!

**Pull request merged**